src/workerd/api/node/tests/http-server-nodejs-test.js

```diff
@@ -0,0 +1,573 @@
+    );
+    server.listen(8080);
+
+    const res = await env.SERVICE.fetch('https://cloudflare.com', {
+      headers: { 'User-Agent': 'node-test' },
+    });
+    strictEqual(res.status, 200);
+  },
+};
+
+// Test is taken from test/parallel/test-http-server-options-server-response.js
+export const testHttpServerOptionsServerResponse = {
+  async test(_ctrl, env) {
+    class MyServerResponse extends http.ServerResponse {
+      status(code) {
+        return this.writeHead(code, { 'Content-Type': 'text/plain' });
+      }
+    }
+
+    await using server = http.createServer(
+      {
+        ServerResponse: MyServerResponse,
+      },
+      (_req, res) => {
+        res.status(200);
+        res.end();
+      }
+    );
+    server.listen(8080);
+
+    const res = await env.SERVICE.fetch('https://cloudflare.com');
+    strictEqual(res.status, 200);
+    strictEqual(res.headers.get('content-type'), 'text/plain');
+  },
+};
+
+// Test is taken from test/parallel/test-http-server-write-after-end.js
+export const testHttpServerWriteAfterEnd = {
+  async test(_ctrl, env) {
+    const { promise, resolve } = Promise.withResolvers();
+    await using server = http.createServer(handle);
+
+    function handle(_req, res) {
+      res.write('hello');
+      res.end();
+
+      queueMicrotask(() => {
+        res.write('world', (err) => {
+          strictEqual(err.code, 'ERR_STREAM_WRITE_AFTER_END');
+          resolve();
+        });
+      });
+    }
+
+    server.listen(8080);
+    await env.SERVICE.fetch('https://cloudflare.com');
+    await promise;
+  },
+};
+
+// Test is taken from test/parallel/test-http-server-write-end-after-end.js
+export const testHttpServerWriteEndAfterEnd = {
+  async test(_ctrl, env) {
+    const { promise, resolve } = Promise.withResolvers();
+    const handle = mock.fn((req, res) => {
+      res.write('hello');
+      res.end();
+
+      queueMicrotask(() => {
+        res.end('world');
+        res.write('world', (err) => {
+          strictEqual(err.code, 'ERR_STREAM_WRITE_AFTER_END');
+          resolve();
+        });
+      });
+    });
+    const server = http.createServer(handle);
+    server.listen(8080);
+
+    await env.SERVICE.fetch('https://cloudflare.com');
+    await promise;
+    strictEqual(handle.mock.callCount(), 1);
+    server.close();
+  },
+};
+
+export const testHandleZeroPortNumber = {
+  async test() {
+    const { promise, resolve } = Promise.withResolvers();
+    const server = http.createServer();
+    const listeningFn = mock.fn();
+    server.on('listening', listeningFn);
+    server.listen(0, () => {
+      ok(server.listening);
+      notStrictEqual(server.port, 0);
+      notStrictEqual(server.address().port, 0);
+      strictEqual(listeningFn.mock.callCount(), 1);
+      server.close();
+      resolve();
+    });
+    await promise;
+  },
+};
+
+export const testInvalidPorts = {
+  async test() {
+    const server = http.createServer();
+    for (const value of [NaN, Infinity]) {
```

Comment:
How about -1, -Infinity, 1.1 (non-integer values), 9999999 (out of range values)